packages/invoices/src/OutboundShipment/DetailView/tabs/GeneralTab.tsx

```diff
@@ -38,22 +80,80 @@ export const GeneralTabComponent: FC<
   GeneralTabProps<OutboundShipmentSummaryItem>
 > = ({ data, columns, onRowClick }) => {
   const { pagination } = usePagination();
-  const t = useTranslation('common');
-  const activeRows = data.filter(({ isDeleted }) => !isDeleted);
+  const { isOn, toggle } = useToggle();
+
+  const [grouped, setGrouped] = React.useState<OutboundShipmentSummaryItem[]>(
+    []
+  );
+  const paged = data.slice(
+    pagination.offset,
+    pagination.offset + pagination.first
+  );
+
+  React.useEffect(() => {
+    const newGrouped: OutboundShipmentSummaryItem[] = [];
+    paged.forEach(row => {
+      const batches = Object.values(row.batches);
+      const lineTotal = (row.sellPricePerPack ?? 0) * (row.numberOfPacks ?? 0);
+
+      if (isOn) {
+        newGrouped.push({
+          ...row,
+          lineTotal,
+          sellPricePerUnit: lineTotal / row.unitQuantity,
+          batch: ifTheSameElseDefault(batches, 'batch', '[multiple]'),
+          expiryDate: ifTheSameElseDefault(batches, 'expiryDate', '[multiple]'),
+          canExpand: Object.keys(row.batches).length > 1,
+        });
```

Comment:
Since this is a summary item, the `batch` and `expiryDate` should already be set like this. The `sellPricePerUnit` and `lineTotal` and `canExpand` field could also just be added to `createSummaryItem` in the reducer file (and then would need to update `recalculateSummary` 🙄 ) but then if `isOn`, you don't do anything?

Reply:
I realised that this morning! will do.